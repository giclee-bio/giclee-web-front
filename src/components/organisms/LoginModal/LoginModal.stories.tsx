import { Meta, StoryObj } from '@storybook/react'

import LoginModal from './LoginModal'

const meta: Meta<typeof LoginModal> = {
  component: LoginModal,
  title: 'LoginModal',
}

export default meta
type Story = StoryObj<typeof LoginModal>

export const Index: Story = {}
