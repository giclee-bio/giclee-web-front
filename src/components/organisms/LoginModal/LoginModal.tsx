'use client'
import { signIn } from 'next-auth/react'

import Button from '@/components/atoms/Button'
import Modal from '@/components/molecules/Modal'

import type { Props as ButtonProps } from '@/components/atoms/Button'
import type { DialogProps } from '@/components/molecules/Modal'

const TriggerButtonProps: ButtonProps = {
  children: 'ログイン',
  type: 'primary',
}

const TriggerDefaultButtonProps: ButtonProps = {
  ...TriggerButtonProps,
  type: 'light',
  width: 'half',
}

const dialogProps: DialogProps = {
  description: '現在は、Discordアカウントとの連携でのみログインが可能です。',
  title: 'giclee.bioへようこそ。',
}

type Props = {
  hero?: boolean
}

const LoginModal: React.FC<Props> = ({ hero }) => {
  return (
    <Modal
      buttonProps={hero ? TriggerButtonProps : TriggerDefaultButtonProps}
      dialogProps={dialogProps}
    >
      {/* ToDo NextAuthに置き換え */}
      <Button onClick={signIn} type='discord'>
        Discordでログイン
      </Button>

      <p className='mt-4 w-64 text-xs'>
        <a className='underline' href='/'>
          利用規約
        </a>
        、
        <a className='underline' href='/'>
          プライバシーポリシー
        </a>
        に同意したうえでログインしてください。
      </p>
    </Modal>
  )
}

export default LoginModal
