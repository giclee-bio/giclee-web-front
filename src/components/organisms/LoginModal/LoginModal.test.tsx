import { render } from '@testing-library/react'

import LoginModal from './LoginModal'

describe('LoginModal', () => {
  it('snapshot', async () => {
    const { asFragment } = render(<LoginModal />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
