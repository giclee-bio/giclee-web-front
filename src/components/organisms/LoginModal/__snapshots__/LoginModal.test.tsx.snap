// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal snapshot 1`] = `
<DocumentFragment>
  <button
    class="h-10 rounded text-center font-bold leading-10 w-32 bg-stone-50 text-black"
  >
    ログイン
  </button>
</DocumentFragment>
`;
